### PR TITLE
samples: modules: canopennode: Add stm32f072b_disco support

### PR DESCRIPTION
**Pull request by mfischer**
Add support for the stm32f072rb discovery board.

Signed-off-by: Moritz Fischer <moritzf@google.com>

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]

File: samples/modules/canopennode/README.rst [hunk 1]
```diff
@@ -61,6 +61,25 @@ The sample can be built and executed for the FRDM-K64F as follows:
 Pressing the button labelled ``SW3`` will increment the button press
 counter object at index ``0x2102`` in the object dictionary.
 
+Building and Running for STM32F072RB Discovery
+==============================================
+The :ref:`stm32f072b_disco_board` board does not come with an onboard CAN
+transceiver. In order to use the CAN bus on the STM32F3 Discovery board, an
```
Comment (erwango) [inline samples/modules/canopennode/README.rst:67]: s/STM32F3/STM32F072RB

File: samples/modules/canopennode/README.rst [hunk 2]
```diff
@@ -61,6 +61,25 @@ The sample can be built and executed for the FRDM-K64F as follows:
...
+external CAN bus tranceiver must be connected to ``PB8`` (``CAN_RX``) and
+``PB9`` (``CAN_TX``). This board supports CANopen LED indicators (red and green
+LEDs)
+
+The sample can be built and executed for the STM32F3 Discovery as follows:
```
Comment (erwango) [inline samples/modules/canopennode/README.rst:72]: s/STM32F3/STM32F072RB

File: samples/modules/canopennode/README.rst [hunk 3]
```diff
@@ -61,6 +61,25 @@ The sample can be built and executed for the FRDM-K64F as follows:
...
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/modules/canopennode
+   :board: stm32f072b_disco
+   :goals: build flash
+   :compact:
+
+Pressing the button labelled ``USER`` will increment the button press counter
+object at index ``0x2102`` in the object dictionary.
```
Comment (erwango) [inline samples/modules/canopennode/README.rst:81]: This is not board specific. It should be removed from each boards instructions and be put into sample description

**Review by erwango — CHANGES_REQUESTED** [review]

File: samples/modules/canopennode/README.rst (hunk 3 above)
Comment (henrikbrixandersen) [reply samples/modules/canopennode/README.rst:81]: Yeah, I need to refactor the sample documentation at some point as more and more board gets added. One thing, though, is that the label for the button _is_ board specific.

**Comment by mfischer** [comment]
FWIW, I updated the example code in the docs with new sample path in a separate PR here: https://github.com/zephyrproject-rtos/zephyr/pull/42021

File: samples/modules/canopennode/README.rst (hunk 1 above)
Comment (mfischer) [reply samples/modules/canopennode/README.rst]: Done.

**Review by erwango — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**